File: Java Spring Calculator/src/main/java/com/softuni/entities/Calculator.java
package com.softuni.entities;

public class Calculator {
    private double leftOperand;
    private double rightOperand;
    private String operator;

    public Calculator(double leftOperand, double rightOperand, String operator) {
        this.leftOperand = leftOperand;
        this.rightOperand = rightOperand;
        this.operator = operator;
    }

    public Calculator() {
    }

    public double getLeftOperand() {
        return leftOperand;
    }

    public double getRightOperand() {
        return rightOperand;
    }

    public String getOperator() {
        return operator;
    }

    public double calculateResult() {
        double result=0;
        switch (operator) {
            case "+":
                result=leftOperand+rightOperand;
                break;
            case "-":
                result=leftOperand-rightOperand;
                break;
            case "*":
                result=leftOperand*rightOperand;
                break;
            case "/":
                result=leftOperand/rightOperand;
                break;
            case "pow":
                result=Math.pow(leftOperand,rightOperand);
                break;
        }
        return result;
    }
}

File: JS Lab Calc/models/Calculator.js
function Calculator(leftOperand,operator,rightOperand){
    this.leftOperand=leftOperand;
    this.operator=operator;
    this.rightOperand=rightOperand;

    this.calculateResult=function () {
        let result=0;
        if (isNaN(this.leftOperand)||isNaN(this.rightOperand)) {
            return 'Invalid Input';
        }
        switch (this.operator) {
            case '+':
                    result=this.leftOperand+this.rightOperand;
                break;
            case '-':
                result=this.leftOperand-this.rightOperand;
                break;
            case '/':
                result=this.leftOperand/this.rightOperand;
                break;
            case '*':
                result=this.leftOperand*this.rightOperand;
                break;
            case'pow':
                result=Math.pow(this.leftOperand*this.rightOperand,2);
                break;
        }
        return result;
    }
}
module.exports=Calculator;